conans/client/command.py

```diff
@@ -299,6 +301,8 @@ def install(self, *args):
             reference = ConanFileReference.loads(args.reference)
         except:
             reference = os.path.normpath(os.path.join(current_path, args.reference))
+            if args.generator:
```

Comment:
Probably not necessary, generators can be useful also when you have a conanfile.txt for many users, and for example one wants to install the YCM generator without modifying the conanfile.txt